src/Adapters/Files/GridFSBucketAdapter.js

```diff
@@ -68,26 +72,71 @@ export class GridFSBucketAdapter extends FilesAdapter {
     const stream = await bucket.openUploadStream(filename, {
       metadata: options.metadata,
     });
-    if (this._encryptionKey !== null) {
-      try {
-        const iv = crypto.randomBytes(16);
-        const cipher = crypto.createCipheriv(this._algorithm, this._encryptionKey, iv);
-        const encryptedResult = Buffer.concat([
-          cipher.update(data),
-          cipher.final(),
-          iv,
-          cipher.getAuthTag(),
-        ]);
-        await stream.write(encryptedResult);
-      } catch (err) {
-        return new Promise((resolve, reject) => {
-          return reject(err);
-        });
+
+    const iv = crypto.randomBytes(16);
+    const cipher = this._encryptionKey !== null
+      ? crypto.createCipheriv(this._algorithm, this._encryptionKey, iv)
+      : null;
+
+    try {
+      // when working with a Blob, it could be over the max size of a buffer, so we need to stream it
+      if (typeof Blob !== 'undefined' && data instanceof Blob) {
```

Comment:
In what circumstances would Blob be undefined?

Reply:
I pulled it from this line in the JS SDK
https://github.com/parse-community/Parse-SDK-JS/blob/3e955cd3944de39a559e38cbac58d86faa892606/src/ParseFile.ts#L119

I think maybe some node versions might not have it?

Reply:
This should not be needed here. In the Parse JS SDK it makes sense, because the environment may not support `Blob` type. In Node.js 15 and later, support for `Blob` was added. Parse Server 7 requires min Node version 18. So this check can be removed.

Reply:
removed.